src/Components/Server/src/Circuits/RemoteNavigationManager.cs

```diff
@@ -64,9 +65,19 @@ public void NotifyLocationChanged(string uri, bool intercepted)
             NotifyLocationChanged(intercepted);
         }
 
+        public ValueTask<bool> HandleLocationChanging(string uri, bool intercepted, bool forceLoad)
+        {
+            return NotifyLocationChanging(uri, intercepted, forceLoad);
+        }
+
         /// <inheritdoc />
-        protected override void NavigateToCore(string uri, bool forceLoad)
+        protected override async void NavigateToCore(string uri, bool forceLoad)
```

Comment:
As we want async/await capabilities when handling a LocationChange I had to modify this function to an ```async void``` (Yikes 🤔)  so too not to introduce breaking changes to the current API Surface. 
The best solution would be to not only change this function to an ```async Task``` , but to also Change ```void Navigationmanager.NavigateTo``` to ```Task Navigationmanager.NavigateTo```. 
So I would like some advice on how we would want to proceed. (The ```WebAssemblyNavigationManager``` has a similar approach )